timemachine/ff/handlers/bonded.py

```diff
@@ -213,4 +217,12 @@ def make_key(idxs):
 
         param_idxs = np.array(param_idxs)
 
-        return params[param_idxs], np.array(improper_idxs, dtype=np.int32)
+        # if no matches found, return arrays that can still be concatenated as expected
```

Comment:
A simulation can be valid even if it contains no improper torsion terms. (E.g. simulation of a water box is valid, even though it contains no torsion terms.)

I think the desired behavior when the force field assigns no improper torsion terms to a mol is to generate potentials with no improper torsions (rather than raise an index error).